src/entities/communitySbt.ts

```diff
@@ -553,80 +539,105 @@ class SBT {
    * ranking of all users. Check if given user is in top 5.
    * If so, assign a top trader/LP badge, otherwise return undefined
    */
-  public async getTopBadge(
+  public async getTopBadges(
     userId: string,
     seasonId: number,
-    isLP: boolean,
     seasonStart: number,
     seasonEnd: number,
     selectedBadgesSubgraphUrl?: string,
-  ): Promise<BadgeResponse | undefined> {
-    const badgeType = getTopBadgeType(seasonId, !isLP);
-    if (!badgeType) return undefined;
+  ): Promise<{
+    topTraderBadge: BadgeResponse | undefined;
+    topLpBadge: BadgeResponse | undefined;
+  }> {
+    const traderBadgeType = getTopBadgeType(seasonId, true);
+    const lpBadgeType = getTopBadgeType(seasonId, false);
 
     if (!selectedBadgesSubgraphUrl || !this.coingeckoKey || !this.ignoredWalletIds) {
-      return undefined;
+      return {
+        topTraderBadge: undefined,
+        topLpBadge: undefined,
+      };
     }
 
     if (!this.ethPrice) {
       this.ethPrice = await geckoEthToUsd(this.coingeckoKey);
     }
 
-    const rankResult = await this.getRanking({
-      seasonStart,
-      seasonEnd,
-      isLP,
-    });
+    const rankResult = await this.getRanking(seasonId);
 
-    for (let rank = 0; rank < 5; rank++) {
-      if (!rankResult[rank]) {
-        return undefined;
-      }
-      const entry = rankResult[rank];
-      if (entry.address === userId) {
-        const badge = await this.constructTopBadge(
+    const traderPosition = rankResult.traderRankResults
+      .slice(0, 5)
+      .find((rank) => rank.address.toLowerCase() === userId.toLowerCase());
+
+    const topTraderBadge = !isUndefined(traderPosition)
+      ? await SBT.constructTopBadge(
           userId,
           seasonId,
           seasonEnd,
-          badgeType,
+          traderBadgeType,
           selectedBadgesSubgraphUrl,
-        );
-        return badge;
-      }
-    }
+        )
+      : undefined;
 
-    return undefined;
-  }
+    const lpPosition = rankResult.lpRankResults
+      .slice(0, 5)
+      .find((rank) => rank.address.toLowerCase() === userId.toLowerCase());
 
-  public async getRanking(args: GetRankingArgs): Promise<RankType[]> {
-    if (!this.subgraphUrl || !this.coingeckoKey || !this.ignoredWalletIds) {
-      return [];
-    }
+    const topLpBadge = !isUndefined(lpPosition)
+      ? await SBT.constructTopBadge(
+          userId,
+          seasonId,
+          seasonEnd,
+          lpBadgeType,
+          selectedBadgesSubgraphUrl,
+        )
+      : undefined;
 
-    if (!this.ethPrice) {
-      this.ethPrice = await geckoEthToUsd(this.coingeckoKey);
+    return {
+      topTraderBadge,
+      topLpBadge,
+    };
+  }
+
+  public async getRanking(seasonId: number): Promise<{
```

Comment:
saw you removed some params of this function, including ignored wallets list. How do we avoid including voltz addresses in the ranking?

Reply:
that variable wasn't used inside the function. What is used is the internal field ``this.ignoredWalletIDs``